### ISSUE TITLE
Resize the Advanced Import Settings window

### ISSUE THREAD

**Issue by YiDoSeon** [godotengine/godot-proposals#10660]
### Describe the project you are working on

3D Game

### Describe the problem or limitation you are having in your project

I have a 49 inch ultra wide monitor. The “Advanced Import Settings” window shown in the screenshot is too large, so I'm reducing its size.
When I open that window again, it goes back to its original size.
![screenshot](https://github.com/user-attachments/assets/00180fb7-361a-44ef-9f07-d49ee898c112)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If this feature was implemented, I wouldn't have to resize that window every time it opened.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

What I want to do is resize the window and when I reopen it, it stays at the resized size.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't know anything about that.

### Is there a reason why this should be core and not an add-on in the asset library?

I don't know if any addons or asset libraries have this functionality, but I believe this is an issue with the editor in the Godot engine.

**Comment by YiDoSeon** [comment]
This applies not only to the Advanced Import Settings dialog but also to most other Import Settings.
I think it would be good if we could solve it together. :)

**Comment by Calinou** [comment]
> I have a 49 inch ultra wide monitor. The “Advanced Import Settings” window shown in the screenshot is too large, so I'm reducing its size.

For this particular issue, I'd probably clamp the aspect ratio of the dialog that spawns by default, so that you automatically get a reasonably sized dialog.

It might still be worth persisting the dialog's position and size across sessions, but we should try to ensure the default size is as usable as possible.

**Comment by YiDoSeon** [comment]
@Calinou Is there any news on this proposal?

**Comment by Calinou** [comment]
> [@Calinou](https://github.com/Calinou?rgh-link-date=2025-03-04T18%3A56%3A25.000Z) Is there any news on this proposal?

Not that I know of.

The dialog's size is controlled here:

https://github.com/godotengine/godot/blob/0e3ffe55c2b1c681d52d1ce91a6ce36c7eb6071b/editor/import/3d/scene_import_settings.cpp#L797

This method accepts an optional parameter to set the ratio of the screen size to use, but width and height can't be chosen separately and we can't enforce a maximum size.

We might be able to get better results with `popup_centered_clamped()` instead, where you can set a minimum size (width and height) but also a fallback ratio that is used as the *maximum* size it can take compared to the screen.

Note that to test this, you may have to create a custom 21:9 (or better, 32:9) resolution on your monitor. On Linux + NVIDIA, you can do it this way:

![Image](https://github.com/user-attachments/assets/5de1f790-e47b-41a8-99b2-c5c32376f7c3)

**Comment by YiDoSeon** [comment]
> > [@Calinou](https://github.com/Calinou?rgh-link-date=2025-03-04T18%3A56%3A25.000Z) Is there any news on this proposal?
> 
> Not that I know of.
> 
> For anyone interested in looking into this, the dialog's size is controlled here:
> 
> https://github.com/godotengine/godot/blob/0e3ffe55c2b1c681d52d1ce91a6ce36c7eb6071b/editor/import/3d/scene_import_settings.cpp#L797
> 
> This method accepts an optional parameter to set the ratio of the screen size to use, but width and height can't be chosen separately and we can't enforce a maximum size.
> 
> We might be able to get better results with `popup_centered_clamped()` instead, where you can set a minimum size (width and height) but also a fallback ratio that is used as the *maximum* size it can take compared to the screen.
> 
> Note that to test this without an ultrawide/superwide monitor, you may have to create a custom 21:9 (or better, 32:9) resolution on your monitor. On Windows, you can use [CRU](https://www.monitortests.com/forum/Thread-Custom-Resolution-Utility-CRU) for this. On Linux + NVIDIA, you can do it this way by clicking **Advanced** at the bottom of the screen configuration dialog:
> 
> ![Image](https://github.com/user-attachments/assets/5de1f790-e47b-41a8-99b2-c5c32376f7c3)

I am a Windows user and my monitor has a resolution of 5120x1440 (32:9).

Does this mean I have to build the godot engine myself to use the popup_centered_clamped() method for testing?

**Comment by Calinou** [comment]
> Does this mean I have to build the godot engine myself to use the popup_centered_clamped() method for testing?

Yes 🙂 See the [instructions](https://docs.godotengine.org/en/stable/contributing/development/compiling/compiling_for_windows.html).

**Comment by YiDoSeon** [comment]
@Calinou 
I tested after building using the popup_centered_clamped() method as you suggested, but the result was not what I wanted.

<img width="1028" alt="Image" src="https://github.com/user-attachments/assets/40022820-b205-46d9-9d6d-42c976d4f4c3" />

**Comment by YiDoSeon** [comment]
I just checked further.
I use popup_centered_clamped() to open the editor and when I open the Advanced Import Settings window, it looks weird like the screenshot.

And if I change the size of the window with the mouse and then open the window again, it stays that size.

But when I reopen the editor, it reverts back to the screenshot.

If the window size is displayed correctly when I open the editor and first open the window, it shouldn't be a problem.

Also, besides “Advanced Import Settings”, “Export Scene to glTF 2.0 File”, “Search Help”, “Search”, etc... when I use a wild resolution monitor, the window opens too wide unnecessarily.

**Comment by YiDoSeon** [comment]
Oh, I missed something
popup_centered_clamped(Size2i(1280, 720));
When I do this, the size of the initially displayed window is fine.

But I'm curious, does using a specific minsize as an argument only affect me?

If I use a specific minsize and it fits, I wonder if it will actually be reflected in the godot release.

If not, do I have to do this again every time a new godot version is released?

**Comment by Calinou** [comment]
> If I use a specific minsize and it fits, I wonder if it will actually be reflected in the godot release.

If you do a custom build, it only affects the custom build in question.

> If not, do I have to do this again every time a new godot version is released?

Yes, you'll need to make a custom build of the editor after updating your copy of the source code to the latest version.

**Comment by YiDoSeon** [comment]
@Calinou 

> Yes, you'll need to make a custom build of the editor after updating your copy of the source code to the latest version.
I'd rather use the official build than a custom build.

Is it possible to add a setting that allows users to specify the size for specific windows in Godot Editor Settings on a per-user basis?
It's hard to find the right value, so it would be nice to use the default at first, but allow users to set their own size in the editor settings.

**Comment by Calinou** [comment]
> Is it possible to add a setting that allows users to specify the size for specific windows in Godot Editor Settings on a per-user basis?

It's better to persist the last value that was used instead of adding an editor setting that won't be tweaked often.

**Comment by YiDoSeon** [comment]
> It's better to persist the last value that was used instead of adding an editor setting that won't be tweaked often. This is precisely the topic of this proposal 🙂

I see.

However, the popup_centred_clamped() method is also eventually reset when the editor is reopened, so does this mean that we need to find a suitable value for the initial value that will be applied every time the editor is opened?

**Comment by Calinou** [comment]
> However, the popup_centred_clamped() method is also eventually reset when the editor is reopened, so does this mean that we need to find a suitable value for the initial value that will be applied every time the editor is opened?

If we persist the last value across sessions, there's no need to modify `poup_centered_clamped()`. This could be done in a separate proposal if there's still a need for it.

**Comment by YiDoSeon** [comment]
> If we persist the last value across sessions, there's no need to modify `poup_centered_clamped()`. This could be done in a separate proposal if there's still a need for it.

It's actually best to keep the last value across sessions, but if that's not possible, it would be nice if the window didn't open too wide at wide resolutions.🙂

So we need to find the right value, right?
